code/frontend/MakeItSo/MakeItSo/Configuration/Services/ConfigurationService.swift

```diff
@@ -0,0 +1,48 @@
+//
+//  ConfigurationService.swift
+//  MakeItSo
+//
+//  Created by Marina Coelho on 19.08.22.
+//  Copyright © 2022 Google LLC. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import Foundation
+import os
+import FirebaseRemoteConfig
+
+public class ConfigurationService: ObservableObject {
+    private let logger = Logger(subsystem: "com.google.firebase.workshop.MakeItSo", category: "configuration")
+    
+    @Published var showDetailsButton: Bool = ConfigurationDefaults.showDetailsButtonValue
+
+    init() {
+        RemoteConfig.remoteConfig().setDefaults(fromPlist: "remote_config_defaults")
+        
+        // Only used for testing, should never deploy this code to production:
+        let settings = RemoteConfigSettings()
+        settings.minimumFetchInterval = 0
+        RemoteConfig.remoteConfig().configSettings = settings
+    }
+    
+    func fetchConfigurationData() {
+        RemoteConfig.remoteConfig().fetch { (status, error) -> Void in
+          if status == .success {
+              RemoteConfig.remoteConfig().activate()
+              self.showDetailsButton = RemoteConfig.remoteConfig().configValue(forKey: ConfigurationDefaults.showDetailsButtonKey).boolValue
+          } else {
+              self.logger.debug("Could not fetch configuration data")
```

Comment:
Might not need to prefix this with self.

Reply:
I tried to remove the prefix, but apparently it's needed :/